api/app/validators/SlotValidator.scala

```diff
@@ -0,0 +1,26 @@
+package validators
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import models.Slot
+import persistence.Store
+
+sealed trait Validation
+final case object Success extends Validation
+final case class Failure(message: String) extends Validation
```

Comment:
Yeah this could definitely be replaced with `Try` as it is! I went with this because I felt like the more expressive names added clarity compared to a more generic `Try` and they map more closely to the outcomes of validation (failure with a message, success with no further info required). I don't know how likely this is, but this also feels like it might be more open to change/flexible in future. I read a couple of articles on the concept of primitive obsession which also influenced thinking here, but I might be missing the spirit of it (or taking it too far!).

I wonder if part of the picture is that this takes quite a bit more code to implement in Scala than the equivalent in something like Haskell (`data Validation = Success | Failure String`) so there's a greater overhead to mentally parsing the code, which I agree is a potential downside.